From c25ea5cbfb155c599fdb0410bea74f6c55af95d7 Mon Sep 17 00:00:00 2001
From: Max Reitz <mreitz@redhat.com>
Date: Sat, 14 Jun 2014 21:41:57 +0200
Subject: [PATCH 20/26] qemu-img: add special exit code if bdrv_check is not supported

RH-Author: Max Reitz <mreitz@redhat.com>
Message-id: <1402782119-29688-2-git-send-email-mreitz@redhat.com>
Patchwork-id: 59223
O-Subject: [RHEL-6.6 qemu-kvm PATCH 1/3] qemu-img: add special exit code if bdrv_check is not supported
Bugzilla: 1054753
RH-Acked-by: Stefan Hajnoczi <stefanha@redhat.com>
RH-Acked-by: Kevin Wolf <kwolf@redhat.com>
RH-Acked-by: Amos Kong <akong@redhat.com>

From: Peter Lieven <pl@kamp.de>

BZ: 1054753

currently it is not possible to distinguish by exitcode if there
has been an error or if bdrv_check is not supported by the image
format. Change the exitcode from 1 to 63 for the latter case.

Signed-off-by: Peter Lieven <pl@kamp.de>
Reviewed-by: Eric Blake <eblake@redhat.com>
Signed-off-by: Kevin Wolf <kwolf@redhat.com>
(cherry picked from commit fefddf951b6dfe51c28d41f86669bfffb68c7a15)

Signed-off-by: Max Reitz <mreitz@redhat.com>
---
 qemu-img.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Signed-off-by: Miroslav Rezanina <mrezanin@redhat.com>
---
 qemu-img.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/qemu-img.c b/qemu-img.c
index b2c78af..5e38dc8 100644
--- a/qemu-img.c
+++ b/qemu-img.c
@@ -572,7 +572,7 @@ static int img_check(int argc, char **argv)
         if (output_format == OFORMAT_HUMAN) {
             error_report("This image format does not support checks");
         }
-        ret = 1;
+        ret = 63;
         goto fail;
     }
 
-- 
1.7.1

